Prepared statement messed up with Oracle

While testing issue #85 I came across the problem with Oracle.
Well, tried to debug, but didn't find anything meaningful.

While running following block with Oracle:

```
    final String query = "select prepstmt_test.col1, prepstmt_test2.col1, prepstmt_test.col2, prepstmt_test2.col2 from prepstmt_test, prepstmt_test2 where prepstmt_test.col2 = prepstmt_test2.col2 and prepstmt_test.col1 = ? and prepstmt_test2.col1 = ?";
    final PreparedStatement prep = getPreparedStatement(query);
    prep.setString(1, "prepstmt_test_col1");
    prep.setString(2, "prepstmt_test_col2");
    final ResultSet rs = prep.executeQuery();
```

I've seen exception:

```
testSameColumnNameInMultipleTables[0: Oracle](com.p6spy.engine.spy.P6TestPreparedStatement)  Time elapsed: 0.73 sec  <<< FAILURE!
java.lang.AssertionError: ORA-00904: "REPSTMT_TEST2"."COL2NDREPSTMT_TEST"."COL1": invalid identifier
 due to error: java.sql.SQLSyntaxErrorException: ORA-00904: "REPSTMT_TEST2"."COL2NDREPSTMT_TEST"."COL1": invalid identifier

    at oracle.jdbc.driver.T4CTTIoer.processError(T4CTTIoer.java:450)
    at oracle.jdbc.driver.T4CTTIoer.processError(T4CTTIoer.java:392)
    at oracle.jdbc.driver.T4CTTIoer.processError(T4CTTIoer.java:385)
    at oracle.jdbc.driver.T4CTTIfun.processError(T4CTTIfun.java:938)
    at oracle.jdbc.driver.T4CTTIfun.receive(T4CTTIfun.java:655)
    at oracle.jdbc.driver.T4CTTIfun.doRPC(T4CTTIfun.java:249)
    at oracle.jdbc.driver.T4C8Odscrarr.doODNY(T4C8Odscrarr.java:96)
    at oracle.jdbc.driver.T4CPreparedStatement.doDescribe(T4CPreparedStatement.java:719)
    at oracle.jdbc.driver.OracleStatement.describe(OracleStatement.java:4223)
    at oracle.jdbc.driver.OracleResultSetMetaData.<init>(OracleResultSetMetaData.java:52)
    at oracle.jdbc.driver.OracleStatement.getResultSetMetaData(OracleStatement.java:4206)
    at oracle.jdbc.driver.OraclePreparedStatement.getMetaData(OraclePreparedStatement.java:4603)
    at oracle.jdbc.driver.OraclePreparedStatementWrapper.getMetaData(OraclePreparedStatementWrapper.java:1510)
    at oracle.jdbc.driver.OracleParameterMetaData.getParameterMetaData(OracleParameterMetaData.java:70)
    at oracle.jdbc.driver.OraclePreparedStatement.getParameterMetaData(OraclePreparedStatement.java:11621)
    at oracle.jdbc.driver.OraclePreparedStatementWrapper.getParameterMetaData(OraclePreparedStatementWrapper.java:1552)
    at com.p6spy.engine.logging.P6LogConnectionPrepareStatementDelegate.invoke(P6LogConnectionPrepareStatementDelegate.java:40)
    at com.p6spy.engine.proxy.GenericInvocationHandler.invoke(GenericInvocationHandler.java:75)
    at net.sf.cglib.proxy.Proxy$ProxyImpl$$EnhancerByCGLIB$$a51e8511.prepareStatement(<generated>)
    at com.p6spy.engine.spy.P6TestPreparedStatement.getPreparedStatement(P6TestPreparedStatement.java:276)
    at com.p6spy.engine.spy.P6TestPreparedStatement.testSameColumnNameInMultipleTables(P6TestPreparedStatement.java:132)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:606)
    at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
    at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
    at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
    at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
    at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
    at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
    at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
    at org.junit.rules.RunRules.evaluate(RunRules.java:20)
    at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
    at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
    at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
    at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
    at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
    at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
    at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
    at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
    at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
    at org.junit.runners.Suite.runChild(Suite.java:127)
    at org.junit.runners.Suite.runChild(Suite.java:26)
    at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
    at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
    at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
    at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
    at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
    at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
    at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:264)
    at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:153)
    at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
    at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
    at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
    at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)

    at org.junit.Assert.fail(Assert.java:88)
    a
```

that is quite weird, because statement seem to be quite messed up. For example: "col2nd" seem to be comming from the "col2."

Maybe Oracle jdbc driver does some magic there. Not sure if it's our misuse, or jdbc bug. 

Debugging didn't show anything relevant. Only thing I've found in the javadocs of driver is existence of: `oracle.jdbc.babelfish .BabelfishPreparedStatement`. Could it somehow come to play and mess it all up? No idea.

To see the bug in action, just remove in the `P6TestPreparedStatement.testSameColumnNameInMultipleTables()` section:

```
if ("Oracle".equals(db)) {
  return;
}
```

--- a/pom.xml
+++ b/pom.xml
@@ -761,8 +761,8 @@
 		<!-- fetched from the cloudbees maven repo -->
  		<dependency>
 			<groupId>com.oracle</groupId>
-	        <artifactId>ojdbc7</artifactId>
-	        <version>12.1.0.1</version>
+	        <artifactId>ojdbc6</artifactId>
+	        <version>11.2.0.4</version>
 	        <scope>test</scope>
 		</dependency>
       </dependencies>

--- a/script/travis/before_install_oracle.sh
+++ b/script/travis/before_install_oracle.sh
@@ -115,5 +115,4 @@ free_restore
 
 # no more needed - stored in the cloudbees mvn repo
 # need to install jdbc to local repo
-# mvn install:install-file -Dfile=/vagrant/script/travis/ojdbc7.jar -DgroupId=com.oracle -DartifactId=ojdbc7 -Dversion=12.1.0.1 -Dpackaging=jar -DgeneratePom=true
-
+#mvn install:install-file -Dfile=/vagrant/script/travis/ojdbc6.jar -DgroupId=com.oracle -DartifactId=ojdbc6 -Dversion=11.2.0.4 -Dpackaging=jar -DgeneratePom=true

--- a/src/test/java/com/p6spy/engine/spy/P6TestPreparedStatement.java
+++ b/src/test/java/com/p6spy/engine/spy/P6TestPreparedStatement.java
@@ -98,12 +98,6 @@ public class P6TestPreparedStatement extends P6TestFramework {
   
   @Test
   public void testSameColumnNameInMultipleTables() throws SQLException {
-
-    // fix pending uncomment for tesing of the:   
-    if ("Oracle".equals(db)) {
-      return;
-    }
-    
     try {
         // insert test data
         {

